File: Power Of 2 and Subsequences.java
class Solution{
    static Long numberOfSubsequences(int N, ArrayList<Long> A){
        // code here
        long m=(long)1e9+7;
        long count=0;
        for(int i=0;i<N;i++){
            long num=A.get(i);
            if((num & num-1)==0){
               count++; 
            }
        }
        long ans=1;
        while(count!=0){
            ans=(ans*2)%m;
            count--;
        }
        return ans-1;
    }
    
}

File: Power Of 2 and Subsequences.py
class Solution:
    def numberOfSubsequences (ob,N,A):
        import math
        A2 = []
        for i in A:
            if math.ceil((math.log(i, 2) )) ==  math.floor((math.log(i, 2))):
                A2.append(i)
        n = len(A2)
        return (2**n - 1) % (10**9+7)
